src/Analyzers/Core/Analyzers/IDEDiagnosticIds.cs

```diff
@@ -171,7 +171,10 @@ internal static class IDEDiagnosticIds
 
         public const string RemoveUnnecessaryLambdaExpressionDiagnosticId = "IDE0200";
 
-        public const string ForEachCastDiagnosticId = "IDE0210";
```

Comment:
intentionally taking this ID.  if we back port this dev17.2, we want to keep these IDEs the same there.